src/main/java/com/epam/training/toto/service/TotoService.java

```diff
@@ -1,4 +1,142 @@
+import java.util.stream.Collectors;
+
 public class TotoService {
+    public static final Integer NUMBER_OF_GAMES = 14;
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy.MM.dd.");
+    private final NumberFormat numberFormat = NumberFormat.getCurrencyInstance(new Locale("uk", "UA"));
+
+    private Map<LocalDate, Round> data;
+
+    public TotoService() {
+        try {
+            data = new CSVParser().readData();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Integer getTheLargestPrize() {
+        List<Integer> allPrizes = data.values()
+                .stream()
+                .map(round -> round.getHits())
+                .flatMap(Collection::stream)
+                .map(hit -> hit.getPrizeForTheGames())
+                .collect(Collectors.toList());
+
+        return Collections.max(allPrizes);
+    }
+
+    private List<Map<Outcome, Long>> getDistributionOfResultsForEachRound() {
+        List<Map<Outcome, Long>> distributionForEachRound = new ArrayList<>(2 * data.size());
+        data.values()
+                .forEach(round -> distributionForEachRound.add(round.getOutcomes()
+                        .stream()
+                        .collect(Collectors.groupingBy(Function.identity(), Collectors.counting())))
+                );
+        return distributionForEachRound;
+    }
+
+    public void printTheLargestPrize() {
+        System.out.println("======== The largest prize: " +
+                numberFormat.format(getTheLargestPrize()) +
+                " ========");
+    }
+
+    public void printDistributionOfTheResults() {
+        System.out.println("\n======== The distribution of the results of each round ========");
+        getDistributionOfResultsForEachRound()
+                .forEach(round -> System.out.println("team #1 won: " +
+                        formatDistribution(round.get(Outcome.FIRST_TEAM_VICTORY)) +
+                        ", team #2 won: " +
+                        formatDistribution(round.get(Outcome.SECOND_TEAM_VICTORY)) +
+                        ", draw: " +
+                        formatDistribution(round.get(Outcome.DRAW)))
+                );
+    }
+
+    private String formatDistribution(Long amountOfVictories) {
+        if (amountOfVictories == null) {
+            return "0%";
+        }
+        return String.format("%.2f ", amountOfVictories / (double) NUMBER_OF_GAMES) + "%";
+    }
+
+    public void printHitsAndAmountForTheWager() {
+        System.out.println("\n======== The hits and amount for the wager ========");
+        Optional<Round> round;
+        do {
+            LocalDate dateToSearch = readDateToSearch();
+            round = getRoundByDate(dateToSearch);
+        } while (!round.isPresent());
+
+        List<Outcome> wager = readWager();
+        int hitCounter = 0;
+        int prize = 0;
+        for (int gameCounter = 0; gameCounter < NUMBER_OF_GAMES; gameCounter++) {
```

Comment:
Is there a possibility to use stream instead for.

Reply:
it compares enum values of two structures
stream features aren't effective in this case
if you have a better solution, please let me know